internal/operands/data-sources/reconcile.go

```diff
@@ -217,66 +205,254 @@ func reconcileEditRole(request *common.Request) (common.ReconcileResult, error)
+		var err error
+		cronByDataSource, err = getCronsByDataSourceMultiArch(d.sourceCollection, &request.Instance.Spec)
+		if err != nil {
+			return dataSourcesAndCrons{}, fmt.Errorf("failed to create DataImportCrons: %w", err)
+		}
+	} else {
+		cronByDataSource = getCronsByDataSource(&request.Instance.Spec)
+	}
+
+	var err error
+	var dataSourceInfos []dataSourceInfo
+	if isMultiarch {
+		dataSourceInfos, err = getDataSourceInfosMultiArch(d.sourceCollection, cronByDataSource, request)
+	} else {
+		dataSourceInfos, err = getDataSourceInfos(d.sourceCollection, cronByDataSource, request)
+	}
+	if err != nil {
+		return dataSourcesAndCrons{}, err
+	}
+
+	for i := range dataSourceInfos {
+		if !dataSourceInfos[i].autoUpdateEnabled {
+			delete(cronByDataSource, client.ObjectKeyFromObject(dataSourceInfos[i].dataSource))
+		}
+	}
+
+	return dataSourcesAndCrons{
+		dataSourceInfos: dataSourceInfos,
+		dataImportCrons: slices.Collect(maps.Values(cronByDataSource)),
+	}, nil
 }
 
-func (d *dataSources) getDataSourcesAndCrons(request *common.Request) (dataSourcesAndCrons, error) {
-	cronTemplates := request.Instance.Spec.CommonTemplates.DataImportCronTemplates
-	cronByDataSource := make(map[client.ObjectKey]*ssp.DataImportCronTemplate, len(cronTemplates))
+func getCronsByDataSource(sspSpec *ssp.SSPSpec) map[client.ObjectKey]cdiv1beta1.DataImportCron {
+	cronTemplates := sspSpec.CommonTemplates.DataImportCronTemplates
+	cronByDataSource := make(map[client.ObjectKey]cdiv1beta1.DataImportCron, len(cronTemplates))
 	for i := range cronTemplates {
-		cron := &cronTemplates[i]
+		cron := cronTemplates[i].AsDataImportCron()
 		if cron.Namespace == "" {
 			cron.Namespace = internal.GoldenImagesNamespace
 		}
-		cronByDataSource[client.ObjectKey{
-			Name:      cron.Spec.ManagedDataSource,
-			Namespace: cron.Namespace,
-		}] = cron
+		// The architecture annotation should not be in the created DataImportCron.
+		delete(cron.Annotations, DataImportCronArchsAnnotation)
+
+		addToCronMap(cronByDataSource, &cron)
+	}
+
+	return cronByDataSource
+}
+
+func getCronsByDataSourceMultiArch(sourceCollection template_bundle.DataSourceCollection, sspSpec *ssp.SSPSpec) (map[client.ObjectKey]cdiv1beta1.DataImportCron, error) {
+	if !ptr.Deref(sspSpec.EnableMultipleArchitectures, false) {
+		return nil, fmt.Errorf("multi-architecture needs to be enabled")
+	}
+
+	clusterArchs, err := architecture.GetSSPArchs(sspSpec)
+	if err != nil {
+		return nil, err
+	}
+
+	cronByDataSource := map[client.ObjectKey]cdiv1beta1.DataImportCron{}
+	cronTemplates := sspSpec.CommonTemplates.DataImportCronTemplates
+	for i := range cronTemplates {
+		originalCron := cronTemplates[i].AsDataImportCron()
+
+		// Need a copy, because it is modified later.
+		cron := originalCron.DeepCopy()
+		if cron.Namespace == "" {
+			cron.Namespace = internal.GoldenImagesNamespace
+		}
+
+		archsAnnotationValue := cron.Annotations[DataImportCronArchsAnnotation]
+		if archsAnnotationValue == "" {
```

Comment:
This should still delete the annotation from the template if it is there but empty, no? e.g. if `ok` then `delete(cron.Annotations, DataImportCronArchsAnnotation)`

Reply:
good point, done